src/main/java/com/programmers/commandline/domain/voucher/repository/impl/VoucherMemoryRepository.java

```diff
@@ -0,0 +1,65 @@
+package com.programmers.commandline.domain.voucher.repository.impl;
+
+import com.programmers.commandline.domain.voucher.entity.Voucher;
+import com.programmers.commandline.domain.voucher.repository.VoucherRepository;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+@Profile({"local"})
+public class VoucherMemoryRepository implements VoucherRepository {
+
+    private List<Voucher> memory = new ArrayList<>();
+
+    public String save(Voucher voucher) {
+        memory.add(voucher);
+        return voucher.getId();
+    }
+
+    @Override
+    public Voucher insert(Voucher voucher) {
+        memory.add(voucher);
+        return voucher;
+    }
+
+    @Override
+    public Voucher update(Voucher voucher) {
+        for (int i = 0; i < memory.size(); i++) {
+            if (memory.get(i).getId().equals(voucher.getId())) {
+                memory.get(i).update(voucher.getDiscount());
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public int count() {
+        return memory.size();
+    }
+
+    @Override
+    public List<Voucher> findAll() {
+        List<Voucher> vouchers = new ArrayList<>();
+        memory.forEach(vouchers::add);
+
+        return vouchers;
+    }
+
+    @Override
+    public Optional<Voucher> findById(String voucherId) {
+        return memory.stream()
+                .filter(voucher -> voucher.getId().equals(voucherId))
+                .findFirst();
+    }
+
+    @Override
+    public void deleteAll() {
+        while (!memory.isEmpty()) {
+            memory.remove(0);
+        }
```

Comment:
```suggestion
        memory.clear();
```
다음과 같이 작성할 수 있을 것 같아요 !

Reply:
정말 유용한 매서드를 하나 배워갑니다.🧚‍♂️